### PR TITLE
Workflow multiple changes

### PR DESCRIPTION
**Pull request by mguikema**

### CODE REVIEW

File: app/apps/workflow/models.py [hunk 1]
```diff
@@ -132,76 +138,83 @@ def wait_for_workflows_and_send_message(message):
 
             # tests if all workflows reached thit point
             if len(workflows_completed) == all_workflows.count() and main_workflow:
+                from .tasks import accept_message_for_workflow
```
Comment (afjlambert) [inline app/apps/workflow/models.py:141]: Why not import at the top?

File: app/apps/workflow/models.py [hunk 2]
```diff
@@ -132,76 +138,83 @@ def wait_for_workflows_and_send_message(message):
...
-                )
-                main_workflow.save(update_fields=["data"])
-                main_workflow.update_workflow()
+                accept_message_for_workflow.delay(main_workflow.id, message, extra_data)
 
                 # delete others
-                other_workflows = all_workflows.exclude(id=main_workflow.id)
-                other_workflows.delete()
+                # other_workflows = all_workflows.exclude(id=main_workflow.id)
+                # other_workflows.delete()
```
Comment (afjlambert) [inline app/apps/workflow/models.py:157]: Why is this commented out? Can it be removed or is it an accidental change?

File: app/apps/workflow/models.py [hunk 3]
```diff
@@ -132,76 +138,83 @@ def wait_for_workflows_and_send_message(message):
...
+        self.set_initial_data(extra_data)
         wf = self._update_workflow(wf)
         wf = self._message(wf, message_name, payload, resultVar, extra_data)
-
         wf = self._update_workflow(wf)
         self.save_workflow_state(wf)
         self._update_db(wf)
 
-    def accept_message(self, message_name):
+    def accept_message(self, message_name, extra_data):
```
Comment (afjlambert) [inline app/apps/workflow/models.py:194]: Why all the print statements? Can we use a logger instead?

File: app/apps/workflow/serializers.py [hunk 4]
```diff
@@ -4,20 +4,22 @@
...
-from .models import Task, Workflow
+from .models import CaseUserTask, CaseWorkflow
 
 
-class TaskSerializer(serializers.ModelSerializer):
+class CaseUserTaskSerializer(serializers.ModelSerializer):
     user_has_permission = serializers.SerializerMethodField()
     camunda_task_id = serializers.CharField(source="id")
     form_variables = serializers.Serializer(source="get_form_variables")
+    # frontend dep: rename to 'task_name'
```
Comment (afjlambert) [inline app/apps/workflow/serializers.py:14]: Not sure what this comment means.

File: app/apps/workflow/tasks.py [hunk 5]
```diff
@@ -1,28 +1,26 @@
...
+def accept_message_for_workflow(self, workflow_id, message, extra_data):
+    from apps.workflow.models import CaseWorkflow
 
-    workflow = Workflow.objects.filter(id=workflow_id).first()
+    workflow = CaseWorkflow.objects.filter(id=workflow_id).first()
     if not workflow:
         return "workflow not found: %s" % workflow_id
-    time.sleep(100)
-    workflow.accept_message(message)
+    # time.sleep(100)
```
Comment (afjlambert) [inline app/apps/workflow/tasks.py:22]: Can you remove this?

File: app/apps/workflow/utils.py [hunk 6]
```diff
@@ -81,14 +81,50 @@ def get_base_path():
     return os.path.dirname(os.path.realpath(__file__))
 
 
-def get_workflow_path(theme_name, workflow_version, workflow_type):
-    return os.path.join(
+def get_latest_version(workflow_type, theme_name="default"):
```
Comment (afjlambert) [inline app/apps/workflow/utils.py:84]: Can you remove / replace the print's?

**Review by afjlambert — CHANGES_REQUESTED** [review]

File: app/apps/workflow/models.py (hunk 1 above)
Comment (mguikema) [reply app/apps/workflow/models.py:141]: Circular import

File: app/apps/workflow/models.py (hunk 2 above)
Comment (mguikema) [reply app/apps/workflow/models.py:157]: Comments are on the way for that

File: app/apps/workflow/models.py (hunk 3 above)
Comment (mguikema) [reply app/apps/workflow/models.py:194]: Local shit, left out in next pr

File: app/apps/workflow/serializers.py (hunk 4 above)
Comment (mguikema) [reply app/apps/workflow/serializers.py:14]: frontend dependency. Refactor stops here for now. Next round of refactor involves frontend to.

File: app/apps/workflow/tasks.py (hunk 5 above)
Comment (mguikema) [reply app/apps/workflow/tasks.py:22]: Next will remove it

File: app/apps/workflow/utils.py (hunk 6 above)
Comment (mguikema) [reply app/apps/workflow/utils.py:84]: Next pr will remove prints

File: app/apps/cases/serializers.py [hunk 7]
```diff
@@ -170,16 +171,25 @@ def create(self, validated_data):
...
+                workflow_instance.message(
+                    "main_process",
+                    settings.DEFAULT_SCHEDULE_ACTIONS[0],
+                    "status_name",
+                    {
+                        "status_name": settings.DEFAULT_SCHEDULE_ACTIONS[0],
+                    },
+                )
+        except DatabaseError:
+            raise serializers.ServiceUnavailable()
```
[the remark this answers is not in the document]
Comment (mguikema) [reply app/apps/cases/serializers.py:192]: This should be an workflow.exception.ServiceUnavailable(), will fix it

**Pull request merged**